docs/DeveloperGuide.md

```diff
@@ -2,37 +2,207 @@
+
+Shown below is a class diagram detailing the interaction between `IngredientList` and `Ingredient`:
+![ingredientClasses.png](img/ClassDiagrams/ingredientsClassDiagram.png)
+
+
+### Command Classes
+The command class deals with all the possible commands accepted by YMFC, with each command representing a specific 
+recognised user input (E.g. ListCommand representing the command to list all the recipes in the recipeList).
 
+#### 1. The `Command` base class
+- Serves as the abstract parent class for all specific commands
+- Defines the basic structure and contract that each command class must follow
+- Each command performs a unique operation and can be expanded to include more functionality as required by YMFC
+- Standardises command behavior and establishes an inheritance hierarchy, allowing for the seamless addition and usage 
+of command functionalities
+- Core Elements:
+  - `isBye`: A flag that tracks if an exit command has been issued, following which programme should be terminated
+    - Set by calling `setBye() `
+    - Checked with `isBye()`
+  - `execute()`: An abstract method that must be implemented by all subclasses
+    - Defines the primary action performed by the command (E.g. add new recipe to `recipeList` for `AddRecipeCommand`)
+
+Shown below is a class diagram detailing the interaction between the parent `Command` class and it's child classes:
+![commandClasses.png](img/ClassDiagrams/commandsClassDiagram.png)
+
+#### 2. Command Flow (Using `Command` Classes)
+1. Initialisation: Instantiate a `Command` subclass by passing user input through the `parseCommand()` of `Parser`
+2. Exit signal: If a command is meant to terminate the app (E.g. `ByeCommand`), set `isBye` to true
+3. Execution: Call `execute()`, performing the desired action and interacting with YMFC's data and UI components
+
+
+#### 3. Adding new `Command` Classes
+- Create a `Command` class:
+  1. Inherit from `Command`: Each new command class extends `Command`, inheriting its structure and providing a specific 
+implementation of the `execute()` method
+  2. Override `execute()`: Implement `execute()` to define how the command should interact with `RecipeList`,
+`IngredientList`, `Ui`, and `Storage`
+  3. Additional methods (Optional): You may add helper methods within the class if necessary
+- Register the command
+  1. Update the `parseCommand()` method in `Parser` to recognise and instantiate the new command
+
+#### Example of `Command` Child Classes
+
+1. `AddRecipeCommand` Class
+The AddRecipeCommand class adds a newly created Recipe object to the currentRecipeList object's ArrayList of Recipes.
+
+Shown below is a sequence diagram detailing how a new recipe is added using the AddRecipeCommand object,
+starting from parsing the user input to the addition of the recipe to the recipeList object,
+to finally printing the "recipe added" message to the User on the CLI:
+![AddRecipes.png](img/SequenceDiagrams/AddRecipesSequenceDiagram.png)
```

Comment:
Would it be possible to make the colours more neutral and easily viewable? Also, for the notations, perhaps you could double check them to ensure that they follow the necessary guidelines?